ml_service/util/attach_compute.py

```diff
@@ -29,14 +24,46 @@ def get_compute(
                 #    vnet_name=vnet_name,
                 #    subnet_name=subnet_name
             )
-            compute_target = ComputeTarget.create(workspace, compute_name,
-                                                  compute_config)
+            compute_target = ComputeTarget.create(
+                workspace, compute_name, compute_config
+            )
             compute_target.wait_for_completion(
-                show_output=True,
-                min_node_count=None,
-                timeout_in_minutes=10)
+                show_output=True, min_node_count=None, timeout_in_minutes=10
+            )
+        return compute_target
+    except ComputeTargetException as e:
+        print(e)
+        print("An error occurred trying to provision compute.")
+        exit(1)
+
+
+def get_compute_for_scoring(workspace: Workspace, compute_name: str, vm_size: str): # NOQA
+    try:
+        if compute_name in workspace.compute_targets:
+            compute_target = workspace.compute_targets[compute_name]
+            if compute_target and type(compute_target) is AmlCompute:
+                print("Found existing compute target " + compute_name + " so using it.") # NOQA
+        else:
+            e = Env()
+            compute_config = AmlCompute.provisioning_configuration(
+                vm_size=vm_size,
+                vm_priority=e.vm_priority_scoring,
+                min_nodes=e.min_nodes_scoring,
+                max_nodes=e.max_nodes_scoring,
+                idle_seconds_before_scaledown="300"
```

Comment:
Are these the only difference between get_compute and get_compute_for_scoring? If so, maybe we pass these as parameters so there's just one get_compute independent of Env.